src/main/java/in/projecteka/consentmanager/user/AuthenticatorService.java

```diff
@@ -0,0 +1,94 @@
+package in.projecteka.consentmanager.user;
+
+import com.google.common.cache.LoadingCache;
+import in.projecteka.consentmanager.user.exception.CacheNotAccessibleException;
+import in.projecteka.consentmanager.user.exception.InvalidSessionException;
+import in.projecteka.consentmanager.user.model.OtpRequest;
+import in.projecteka.consentmanager.user.model.OtpVerification;
+import in.projecteka.consentmanager.user.model.TemporarySession;
+import in.projecteka.consentmanager.user.model.Token;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import org.springframework.util.StringUtils;
+
+import java.util.*;
+import java.util.concurrent.ExecutionException  ;
+import java.util.function.Function;
+
+public class AuthenticatorService {
+
+    private JWTProperties jwtProperties;
+    private LoadingCache<String, Optional<String>> sessionCache;
+    private LoadingCache<String, Optional<String>> secondSessionCache;
+    public static final long JWT_TOKEN_VALIDITY = 5 * 60 * 60;
+
+    public AuthenticatorService(JWTProperties jwtProperties,
+                                LoadingCache<String, Optional<String>> sessionCache,
+                                LoadingCache<String, Optional<String>> secondSessionCache) {
+        this.jwtProperties = jwtProperties;
+        this.sessionCache = sessionCache;
+        this.secondSessionCache = secondSessionCache;
+    }
+
+    public TemporarySession cacheAndSendSession(OtpRequest requestBody) {
```

Comment:
Can we have a map method to create temporary session outside, and just have set method with temporary session id and mobilenumber?